packages/admin/cms-admin/src/dam/Table/license/getLicenseValidityInformation.tsx

```diff
@@ -0,0 +1,61 @@
+import { add, compareAsc, getUnixTime } from "date-fns";
+
+const currentDate = new Date();
+const thirtyDaysInSeconds = 60 * 60 * 24 * 30;
+
+interface GetLicenseValidityInformationParams {
+    durationFrom?: Date;
+    durationTo?: Date;
+}
+
+interface LicenseValidityInformation {
+    isNotValidYet: boolean;
+    expiresSoon: boolean;
+    isExpired: boolean;
+    isValid: boolean;
+    startDate?: Date;
+    expirationDate?: Date;
+}
+
+export const getLicenseValidityInformation = ({ durationFrom, durationTo }: GetLicenseValidityInformationParams): LicenseValidityInformation => {
```

Comment:
I'd prefer if this information is calculated in the API. Might be necessary for controllers providing the image file to check if it is still valid. You could add a field resolver to the license type to achieve this